pilot/pkg/networking/core/v1alpha3/tls.go

```diff
@@ -146,7 +146,14 @@ func buildSidecarOutboundTLSFilterChainOpts(env *model.Environment, node *model.
 
 	// HTTPS or TLS ports without associated virtual service will be treated as opaque TCP traffic.
 	if !hasTLSMatch {
-		clusterName := model.BuildSubsetKey(model.TrafficDirectionOutbound, "", service.Hostname, listenPort.Port)
+		port := listenPort.Port
+
+		// If the service has only one port, use that instead of the listenPort
```

Comment:
?? this is a natural fallout of the sidecar API. you can listen on one port but be forwarding to a set of single-port services. No point in forcing user to create same port as the service port. We can infer it. This is how we have been doing the port selection in virtual services for the last 8 months (in both gateways and sidecars). The reason this has been added here is because this code is no longer being triggered by just service-port based listeners. Its also being triggered by sidecar config's listeners that could be on a totally different port or a unix domain socket